controllers/admin/AdminAjaxPs_buybuttonliteController.php

```diff
@@ -0,0 +1,118 @@
+            return false;
+        }
+        $results_array = array();
+        foreach ($result as $row) {
+            if ($this->hasCombinations($row['id_product'])) {
+                $row['attribute_name'] = $this->getAttributes($row['id_product_attribute'], $id_lang);
+                $row['image_link'] = $this->getProductAttributeImage($row['id_product'], $row['id_product_attribute']);
+            } else {
+                $row['image_link'] = $this->getProductImage($row['id_product']);
+            }
+            $results_array[] = $row;
+        }
+        $this->content = json_encode($results_array);
+    }
+
+    public function getProductImage($id_product)
+    {
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getCover($id_product);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function getProductAttributeImage($id_product, $id_product_attribute)
+    {
+        $context = Context::getContext();
+        $id_lang = $context->language->id;
+        $id_shop = $context->shop->id;
+
+        $product = new Product($id_product);
+        $link_rewrite = $product->link_rewrite;
+        $link = new Link();
+        $id_image = Image::getBestImageAttribute($id_shop, $id_lang, $id_product, $id_product_attribute);
+        $id_image = $id_image['id_image'];
+
+        $image_link = $link->getImageLink($link_rewrite, $id_image, ImageType::getFormattedName('small'));
+
+        return Tools::getProtocol().$image_link;
+    }
+
+    public function hasCombinations($id_product)
+    {
+        if (is_null($id_product) || 0 >= $id_product) {
+            return false;
+        }
+        $attributes = Product::getAttributesInformationsByProduct($id_product);
+        return !empty($attributes);
+    }
+
+    public function getAttributes($id_product_attribute, $id_lang)
+    {
+        $sql = 'SELECT agl.name as label, al.name as value
+            FROM ' . _DB_PREFIX_ . 'product_attribute_combination pac
+            LEFT JOIN ' . _DB_PREFIX_ . 'attribute a ON (a.id_attribute = pac.id_attribute)
+            LEFT JOIN ' . _DB_PREFIX_ . 'attribute_lang al ON (al.id_attribute = a.id_attribute AND al.id_lang=' . (int) $id_lang . ')
+            LEFT JOIN ' . _DB_PREFIX_ . 'attribute_group ag ON (ag.id_attribute_group = a.id_attribute_group)
+            LEFT JOIN ' . _DB_PREFIX_ . 'attribute_group_lang agl ON (agl.id_attribute_group = ag.id_attribute_group AND agl.id_lang=' . (int) $id_lang . ')
+            WHERE pac.id_product_attribute=' . (int) $id_product_attribute;
+
+        $results = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql);
+
+        foreach ($results as $attribute) {
+            $attributes[] = implode($attribute, ' - ');
```

Comment:
Can you initialize this `$attributes` as an empty array ? It is better of `$results` is empty

```
$attributes = [];
foreach ($results as $attribute) {
...
```